nncf/openvino/quantization/helpers.py

```diff
@@ -0,0 +1,106 @@
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import tempfile
+from pathlib import Path
+from typing import Optional
+
+import openvino.runtime as ov
+from openvino.tools import pot
+
+from nncf.data import DataLoader
+from nncf.openvino.utils import POTDataLoader
+from nncf.openvino.engine import OVEngine
+from nncf.quantization.params import TargetDevice
+from nncf.common.quantization.structs import QuantizationPreset
+
+
+def _convert_openvino_model_to_compressed_model(model: ov.Model,
+                                                target_device: str) -> pot.graph.nx_model.CompressedModel:
+    """
+    Serializes the provided OpenVINO model and loads the model in the POT representation.
+
+    :param model: The OpenVINO model.
+    :param target_device: The target device.
+    :return: The POT representation of the provided model.
+    """
+    with tempfile.TemporaryDirectory(dir=tempfile.gettempdir()) as tmp_dir:
+        xml_path = str(Path(tmp_dir).joinpath('model.xml'))
+        bin_path = str(Path(tmp_dir).joinpath('model.bin'))
+        ov.serialize(model, xml_path, bin_path)
+        model_config = {
+            'model_name': 'model',
+            'model': xml_path,
+            'weights': bin_path,
+        }
+        pot_model = pot.load_model(model_config, target_device)
+
+    return pot_model
+
+
+def _convert_compressed_model_to_openvino_model(model: pot.graph.nx_model.CompressedModel) -> ov.Model:
+    """
+    Saves the provided POT compressed model and loads it as `openvino.runtime.Model` object.
+
+    :param model: The POT compressed model.
+    :return: The `openvino.runtime.Model`  object which represents the provided model.
+    """
+    with tempfile.TemporaryDirectory(dir=tempfile.gettempdir()) as tmp_dir:
+        paths = pot.save_model(model, save_path=tmp_dir, model_name='model')
+        xml_path = paths[0]['model']
+        bin_path = paths[0]['weights']
+        ie = ov.Core()
+        ov_model = ie.read_model(xml_path, bin_path)
+    return ov_model
+
+
+def quantize_impl(model: ov.Model,
+                  calibration_dataset: DataLoader,
+                  preset: QuantizationPreset,
+                  target_device: TargetDevice,
+                  subset_size: int,
+                  fast_error_correction: bool,
+                  model_type: Optional[str] = None) -> ov.Model:
+    """
+    Implementation of the `quantize()` method for the OpenVINO backend.
+    """
+    pot_model = _convert_openvino_model_to_compressed_model(model, target_device)
+
+    engine_config = {
+        'device': 'CPU',
+        'stat_requests_number': 2,
+        'eval_requests_number': 2,
```

Comment:
BTW, why is the `stat_requests_number` and `eval_requests_number` set as 2?

Reply:
As said @AlexKoff88: "Based on the recent experience, I would recommend setting the number of requests to 2 at least".

Reply:
@KodiaqQ Do you have any concerns?

Reply:
For some types of networks (for example, GNA-specific), such parameters will affect the accuracy due to ReadValue-Assign pairs. But since the GNA isn't supported here, I don't know, should we care about this now.

Reply:
I think this case will be covered when we will have added the `advance_parameters: AdvancedQuantizationParameters` argument. So, we will be able to add some backend-specific parameters to be configured inside this class.